lib/liquid/combinators/tags/case.ex

```diff
@@ -23,6 +23,13 @@ defmodule Liquid.Combinators.Tags.Case do
   alias Liquid.Combinators.{Tag, General}
   alias Liquid.Combinators.Tags.Generic
 
+  @type t :: [
+          case: [
+            variable: LexicalToken.value(),
+            clauses: [String.t() | [when: [conditions: [body: Liquid.t()]]]]
```

Comment:
Based on the tests, I believe this type is incorrect. The conditions aren't a list of tagged :body tuples.
I also think this could be semantically improved to:
```
clauses: [String.t() | [when: condition, then: Liquid.t()]]
```
Not that condition is singular.

Reply:
You're right, condition must be defined as a new type. I'm going to discuss with Samuel how we can define the `condition` type. 

My first idea when changing `body` for `then` was "I like it" but now I think that I prefer the consistency of `body` tag. Every tag which returns a `Liquid.t()` has a `body` (maybe `block` could be more appropriate)